### ISSUE TITLE
vcpkg-tool-meson: PKG_CONFIG_PATH is no longer respected

### ISSUE THREAD

**Issue by marcbull** [microsoft/vcpkg#40305]
**Describe the bug**
The PR https://github.com/microsoft/vcpkg/pull/38901 introduced a problem with meson. On my system I have a library installed in `/usr/local` and I'm using a custom `PKG_CONFIG_PATH` to find the library. Prior to 2024.06.15 the library was found by vcpkg.

After some research, the PR added a command line argument to run meson, which should expand the list of additional paths for pkg-config to search. However, it also prevents meson from using the `PKG_CONFIG_PATH`. I verified this by creating a simple meson.build, searching for the library in `/usr/local`, setting the `PKG_CONFIG_PATH` environment variable, and running meson with and without the `--pkg-config-path` argument.

**Environment**
- OS: Linux

**Expected behavior**
Paths in the PKG_CONFIG_PATH environment variable should be respected when calling `vcpkg_configure_meson()`.

**Comment by JonLiu1993** [comment]
@Neumann-A, could you please take a look?

**Comment by Neumann-A** [comment]
> @Neumann-A, could you please take a look?

Yeah is missing feal free to restore it.

**Comment by dg0yt** [comment]
You could also provide a suitable `pkg-config` wrapper in the `PATH`.

The general problem with the environment variables is that vcpkg doesn't know if they are for the host or for the target.
For triplet awareness, you could set cmake variable `PKGCONFIG` to a suitable wrapper in the triplet file.

**Comment by marcbull** [comment]
> You could also provide a suitable `pkg-config` wrapper in the `PATH`.

Yes, this is a possible solution. But I also want to point out that this behavior might be confusing for other users.

> The general problem with the environment variables is that vcpkg doesn't know if they are for the host or for the target. For triplet awareness, you could set cmake variable `PKGCONFIG` to a suitable wrapper in the triplet file.

So this even extends even further to `LD_LIBRARY_PATH`, `LDFLAGS`, `CFLAGS`, `CXXFLAGS`, ...

**Comment by dg0yt** [comment]
> So this even extends even further to `LD_LIBRARY_PATH`, `LDFLAGS`, `CFLAGS`, `CXXFLAGS`, ...

Not `LD_LIBRARY_PATH`. But yes. And "this behavior might be confusing for other users".
(I sometimes overwrite `CXX` with clang++-12.)
